lib/client/api.go

```diff
@@ -469,15 +469,6 @@ func RetryWithRelogin(ctx context.Context, tc *TeleportClient, fn func() error)
 		log.Warningf("Failed to save profile: %v", err)
 		return trace.Wrap(err)
 	}
-	// Override client's auth methods, current cluster and user name
-	authMethod, err := key.AsAuthMethod()
-	if err != nil {
-		return trace.Wrap(err)
-	}
-	// After successful login we have local agent updated with latest
-	// and greatest auth information, setup client to try only this new
-	// method fetched from key, to isolate the retry
-	tc.Config.AuthMethods = []ssh.AuthMethod{authMethod}
```

Comment:
Why did you remove this?

Reply:
This messes up the relogin flow.

`tc.Config.AuthMethods` is used as the first choice when connecting to a proxy over SSH.
If you were originally connecting to a leaf cluster (as in `tsh ssh -J leaf.proxy.com somenode`) and relogin kicked in, you're be logged in via the root cluster (which is correct).
But after you've re-logged through the root cluster `tc.Config.AuthMethods` would have an SSH cert for the root cluster, not for the leaf.
The cert caching and loading logic that @andrejtokarcik added correctly adds a leaf cert into the `localAgent`, but it gets overriden by `tc.Config.AuthMethods`.